plugins/module_utils/mh/module_helper_dest_file.py

```diff
@@ -0,0 +1,213 @@
+# -*- coding: utf-8 -*-
+# (c) 2022, DEMAREST Maxime <maxime@indelog.fr>
+# Copyright: (c) 2020, Ansible Project
+# Simplified BSD License (see licenses/simplified_bsd.txt or
+# https://opensource.org/licenses/BSD-2-Clause)
+"""
+Toolbox that help to do somme basics on a destination file for module that need
+it.
+"""
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+import os
+import tempfile
+
+from abc import abstractmethod
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible_collections.community.general.plugins.module_utils.mh.module_helper import ModuleHelper
+from ansible_collections.community.general.plugins.module_utils.mh.exceptions import ModuleHelperException
+from ansible_collections.community.general.plugins.module_utils.mh.deco import (
+    cause_changes,
+    check_mode_skip)
+
+
+class DestNotExists(ModuleHelperException):
+    pass
+
+
+class ParentNotWriteable(ModuleHelperException):
+    pass
+
+
+class DestNotReadable(ModuleHelperException):
+    pass
+
+
+class DestNotWriteable(ModuleHelperException):
+    pass
+
+
+class DestNotRegularFile(ModuleHelperException):
+    pass
+
+
+class CantCreateBackup(ModuleHelperException):
+    pass
+
+
+def check_if_dest_exists(dest):
```

Comment:
I just wanted to keep this logic separate to avoid overloading the class `DestFileModuleHelper` with this and kept these function reusable in reusable in other context.